share/functions/__fish_man_page.fish

```diff
@@ -1,4 +1,16 @@
 function __fish_man_page
-    man (basename (commandline -po; echo)[1]) ^/dev/null
-    or printf \a
+    # Get all commandline tokens not starting with "-"
+    set -l args (string join \n -- (commandline -po) | grep -v '^-')
```

Comment:
What is the `string join` supposed to do here?

Also, the grep can be replaced with `string match`.

So: `commandline -po | string match -rv '^-'` should work.